plugins/extractors/maxcompute/maxcompute.go

```diff
@@ -0,0 +1,323 @@
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Config struct {
+	ProjectName     string `mapstructure:"project_name"`
+	EndpointProject string `mapstructure:"endpoint_project"`
+	AccessKey       struct {
+		ID     string `mapstructure:"id"`
+		Secret string `mapstructure:"secret"`
+	} `mapstructure:"access_key"`
+	SchemaName string `mapstructure:"schema_name,omitempty"`
+	Exclude    struct {
+		Schemas []string `mapstructure:"schemas"`
+		Tables  []string `mapstructure:"tables"`
+	} `mapstructure:"exclude,omitempty"`
+	Concurrency int `mapstructure:"concurrency,omitempty"`
+}
+
+type Extractor struct {
+	plugins.BaseExtractor
+	logger log.Logger
+	config Config
+
+	client    Client
+	newClient NewClientFunc
+	eg        *errgroup.Group
+}
+
+type NewClientFunc func(ctx context.Context, logger log.Logger, config Config) (Client, error)
+
+//go:embed README.md
+var summary string
+
+var sampleConfig = `
+project_name: goto_test
+endpoint_project: https://service.ap-southeast-5.maxcompute.aliyun.com/api
+access_key:
+    id: access_key_id
+    secret: access_key_secret
+schema_name: default
+exclude:
+	schemas:
+    	- schema_a
+    	- schema_b
+    tables:
+    	- schema_c.table_a
+concurrency: 10
+`
+
+var info = plugins.Info{
+	Description:  "MaxCompute metadata and metrics",
+	SampleConfig: sampleConfig,
+	Tags:         []string{"alicloud", "table"},
+	Summary:      summary,
+}
+
+//go:generate mockery --name=Client -r --case underscore --with-expecter --structname MaxComputeClient --filename maxcompute_client_mock.go --output=./mocks
+type Client interface {
+	ListSchema(ctx context.Context) ([]*odps.Schema, error)
+	ListTable(ctx context.Context, schemaName string) ([]*odps.Table, error)
+	GetTableSchema(ctx context.Context, table *odps.Table) (string, *tableschema.TableSchema, error)
+}
+
+func New(logger log.Logger, clientFunc NewClientFunc) *Extractor {
+	e := &Extractor{
+		logger:    logger,
+		newClient: clientFunc,
+	}
+	e.BaseExtractor = plugins.NewBaseExtractor(info, &e.config)
+	e.ScopeNotRequired = true
+
+	return e
+}
+
+func (e *Extractor) Init(ctx context.Context, config plugins.Config) error {
+	if err := e.BaseExtractor.Init(ctx, config); err != nil {
+		return err
+	}
+
+	if e.config.ProjectName == "" {
+		return fmt.Errorf("project_name is required")
+	}
+	if e.config.AccessKey.ID == "" || e.config.AccessKey.Secret == "" {
+		return fmt.Errorf("access_key is required")
+	}
+	if e.config.EndpointProject == "" {
+		return fmt.Errorf("endpoint_project is required")
+	}
+	if e.config.Concurrency == 0 {
+		e.config.Concurrency = 1
+	}
+
+	var err error
+	e.client, err = e.newClient(ctx, e.logger, e.config)
+	if err != nil {
+		return err
+	}
+
+	e.eg = &errgroup.Group{}
+	e.eg.SetLimit(e.config.Concurrency)
+
+	return nil
+}
+
+func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) error {
+	schemas, err := e.client.ListSchema(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
since we'll still process all succeed schema, then the condition should be `if err != nill && len(schemas) == 0`

Reply:
addressed